api/src/tailer/mod.rs

```diff
@@ -0,0 +1,347 @@
+use std::process::Stdio;
+use std::sync::Arc;
+
+use bytes::{Buf, BytesMut};
+use combine::parser::byte::crlf;
+use combine::{
+    error::{ParseError, StreamError},
+    none_of,
+    parser::{
+        combinator::{any_partial_state, AnyPartialState},
+        range::recognize,
+    },
+    skip_many,
+    stream::{easy, PartialStream, RangeStream, StreamErrorFor},
+    Parser,
+};
+use futures::{Stream, StreamExt};
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio::process::ChildStdin;
+use tokio::sync::Mutex;
+use tokio_util::codec::{Decoder, FramedRead};
+use tracing::{error, info, trace, warn};
+#[cfg(windows)]
+use win32job::Job;
+
+use http::types::body::LineBuilder;
+
+use crate::tailer::error::TailerError;
+
+mod error;
+
+pub struct TailerApiDecoder {
+    state: AnyPartialState,
+    _stdin: ChildStdin,
+    #[cfg(windows)]
+    _job: Job,
+}
+
+type TailerRecord = String;
+
+// The actual parser for the Tailer API format
+fn decode_parser<'a, Input>(
+) -> impl Parser<Input, Output = TailerRecord, PartialState = AnyPartialState> + 'a
+where
+    Input: RangeStream<Token = u8, Range = &'a [u8]> + 'a,
+    Input::Error: ParseError<Input::Token, Input::Range, Input::Position>,
+{
+    /*
+    Tailer API log lines are 'etf-8' encoded and separated by a single newline.
```

Comment:
When you say tailer API lines are encoded this way - but eventually customers will be able to write their own tailers - perhaps we should add some error handling around this as we can't guarantee how an outside source will format their data.

Reply:
We do lossy string ending in agent on data path to backend (through http etc). Meaning if encoding is wrong it will be mitigated.